Library/Homebrew/formula_auditor.rb

```diff
@@ -391,7 +391,7 @@ def audit_conflicts
                   "canonical name (#{conflicting_formula.name}) instead of #{conflict.name}"
         end
 
-        reverse_conflict_found = false
+        reverse_conflict_found = T.let(false, T::Boolean)
```

Comment:
This is necessary to allow the type to widened from `FalseClass` to `T.any(FalseClass, TrueClass)` (ruby does not have a native notion of a `Boolean` type). This specific example is covered in the [docs](https://sorbet.org/docs/error-reference#7001) for the ensuing error. It's also touched on in the type annotations [section](https://sorbet.org/docs/why-type-annotations#couldnt-sorbet-at-least-special-case-true-and-false).

Sorry for not elaborating on this PR, there are already ~22 uses of this boolean lvar pattern on `master`, so I thought it might already be understood.

Reply:
No worries, don't expect you to preemptively elaborate.

This is one of the things I dislike about Sorbet, it feels weird that it's not more intuitive/permissive on Boolean usage like this.

Reply:
`nil` will just be `NilClass`, `T.nilable(Version)` is basically `T.any(NilClass, Version)`.

Reply:
Fair, although I would characterize this as a bug in Ruby. (Mats is entrenched his no-Boolean-class position). The second link above gives the rationale for why Sorbet doesn’t special-case true/false values.